dask_planner/src/expression.rs

```diff
@@ -224,13 +252,13 @@ impl PyExpr {
                 right: _,
             } => Ok(format!("{}", op)),
             Expr::ScalarFunction { fun, args: _ } => Ok(format!("{}", fun)),
-            Expr::Cast {
-                expr: _,
-                data_type: _,
-            } => Ok(String::from("cast")),
-            _ => Err(PyErr::new::<pyo3::exceptions::PyTypeError, _>(
-                "Catch all triggered ....",
-            )),
+            Expr::Cast { .. } => Ok(String::from("cast")),
+            Expr::Between { .. } => Ok(String::from("between")),
+            Expr::Case { .. } => Ok(String::from("case")),
```

Comment:
minor nit: original code is absolutely fine but it is more typical to just use `.to_string()` or `.to_owned()`.

```suggestion
            Expr::Cast { .. } => Ok("cast".to_string()),
            Expr::Between { .. } => Ok("between".to_string()),
            Expr::Case { .. } => Ok("case".to_string()),
```